### PR TITLE
Apply black formatting to the whole project and add tools to enforce it

### LINKED ISSUE #700: Reformat codebase using Black
- [ ] Land currently in-flight work
    - [ ] #688 
    - [x] #690 
    - [x] #697  
    - [x] @OddBloke's local branch
- [x] Create pyproject.toml to configure line length of 79
- [x] Reformat codebase with Black
- [x] Add tox target to ensure that codebase is compliant
- [x] Optional: Add pre-commit hook to make non-compliance easier for developers to notice

### PR DESCRIPTION
**Pull request by OddBloke**
Fixes #700

### CODE REVIEW

**Comment by OddBloke** [comment]
I've realised we probably want docs for this; I'm going to submit those separately, so we can land this ASAP so new code doesn't hit horrible conflicts.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+repos:
+  - repo: https://github.com/ambv/black
+    rev: 19.3b0
```
Comment (blackboxsw) [inline .pre-commit-config.yaml]: Can you add a an anchor comment here pointing to dev-requirements.txt too

Comment (OddBloke) [reply .pre-commit-config.yaml]: Done.

File: uaclient/entitlements/base.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 try:
     from typing import Any, Callable, Dict, Optional, Tuple  # noqa: F401
+
```
[the remark this answers is not in the document]
Comment (OddBloke) [reply uaclient/entitlements/base.py]: Yep, black puts it back if I remove it.

File: uaclient/entitlements/livepatch.py [hunk 3]
```diff
@@ -25,14 +25,17 @@ class LivepatchEntitlement(base.UAEntitlement):
     name = 'livepatch'
     title = 'Livepatch'
     description = (
-        'Canonical Livepatch Service'
-        ' (https://ubuntu.com/livepatch)')
+        'Canonical Livepatch Service' ' (https://ubuntu.com/livepatch)'
```
[the remark this answers is not in the document]
Comment (OddBloke) [reply uaclient/entitlements/livepatch.py]: Fixed.

**Pull request merged**